pkg/platform/aws/aws.go

```diff
@@ -0,0 +1,194 @@
+				iface.Mac = macAddr
+			}
+			iface.LinkSpeed = a.hostHelpers.GetNetDevLinkSpeed(name)
+			iface.LinkType = a.hostHelpers.GetLinkType(name)
+		}
+		if len(iface.VFs) != 1 {
+			log.Log.Error(nil, "only one vf should exist", "iface", iface)
+			return nil, fmt.Errorf("unexpected number of vfs found for device %s", iface.Name)
+		}
+		iface.VFs[0] = sriovnetworkv1.VirtualFunction{
+			PciAddress: iface.PciAddress,
+			Driver:     driver,
+			VfID:       0,
+			Vendor:     iface.Vendor,
+			DeviceID:   iface.DeviceID,
+			Mtu:        iface.Mtu,
+			Mac:        iface.Mac,
+		}
+		a.loadedDevicesInfo[idx] = iface
+	}
+	return a.loadedDevicesInfo, nil
+}
+
+// DiscoverBridges is not supported on AWS platform.
+// Returns ErrOperationNotSupportedByPlatform as AWS does not support software bridge management.
+func (a *Aws) DiscoverBridges() (sriovnetworkv1.Bridges, error) {
+	if vars.ManageSoftwareBridges {
+		return sriovnetworkv1.Bridges{}, vars.ErrOperationNotSupportedByPlatform
+	}
+	return sriovnetworkv1.Bridges{}, nil
+}
+
+// createDevicesInfo creates the AWS device info map
+// This function is used to create the AWS device info map from the metadata server.
+func (a *Aws) createDevicesInfo() error {
+	funcLog := log.Log.WithName("getDataFromMetadataService")
+	a.loadedDevicesInfo = make(sriovnetworkv1.InterfaceExts, 0)
+	funcLog.Info("getting aws network info from metadata server")
+	metaData, err := a.hostHelpers.HTTPGetFetchData(metadataBaseURL + macsPath)
+	if err != nil {
+		return fmt.Errorf("error getting aws meta_data from %s: %v", metadataBaseURL+macsPath, err)
+	}
+
+	// If the endpoint returns an empty body (e.g., no MACs available or an issue),
+	// treat it as no MACs found rather than an error. The caller can then decide how to handle an empty list.
+	if metaData == "" {
+		return nil
+	}
+
+	// MAC addresses are returned separated by newlines, each ending with a '/'.
+	// Example raw response: "0e:1a:95:aa:12:a1/\n0e:92:d3:ee:52:1b/"
+	macEntries := strings.Split(metaData, "\n")
+	if len(macEntries) == 0 {
+		return nil
+	}
+
+	macAddressToSubNetID := map[string]string{}
+	for _, macEntry := range macEntries {
+		subnetIDURL := metadataBaseURL + macsPath + macEntry + subnetIDSuffix
+		subnetIDData, err := a.hostHelpers.HTTPGetFetchData(subnetIDURL)
+		if err != nil {
+			return fmt.Errorf("error getting aws subnet_id from %s: %v", subnetIDURL, err)
+		}
+
+		if subnetIDData == "" {
+			return fmt.Errorf("empty subnet_id from %s: %v", subnetIDURL, err)
+		}
+		macAddressToSubNetID[strings.ReplaceAll(macEntry, "/", "")] = subnetIDData
+	}
+
+	pfList, err := a.hostHelpers.DiscoverSriovVirtualDevices()
+	if err != nil {
+		return err
+	}
+
+	for _, iface := range pfList {
+		subnetID, exist := macAddressToSubNetID[iface.Mac]
+		if !exist {
+			continue
+		}
+
+		subnetID = strings.TrimPrefix(subnetID, "subnet-")
+		iface.NetFilter = fmt.Sprintf("aws/NetworkID:%s", subnetID)
```

Comment:
not related to this but i saw we have `OpenstackNetworkID` defined in `api/v1/helper.go` perhaps its worth moving it to openstack platform impl.

also perhaps we should have aws/NetworkID as const in the top of the file ?